### PR TITLE
Core: Support libgit2 v1 when using system libgit2

### PR DESCRIPTION
**Pull request by flokli**
This ifdefs doesn't properly handle libgit2 >=1.0
(where LIBGIT2_VER_MAJOR == 1)

Invert the logic like done in other places.

### Describe the pull request:

- [x] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### CODE REVIEW

File: core/git-access.c [hunk 1]
```diff
@@ -359,10 +359,10 @@ static int try_to_git_merge(git_repository *repo, git_reference **local_p, git_r
 	}
 
 	git_merge_init_options(&merge_options, GIT_MERGE_OPTIONS_VERSION);
-#if !LIBGIT2_VER_MAJOR && LIBGIT2_VER_MINOR > 23
-	merge_options.flags = GIT_MERGE_FIND_RENAMES;
-#else
+#if !LIBGIT2_VER_MAJOR && LIBGIT2_VER_MINOR < 22
```
Comment (dirkhh) [inline core/git-access.c:362]: So this is obviously incorrect as it switches from `>23` to in effect `!<22` which is very wrong.
But the bigger question is "why would we still support anything older than libgit2 0.26.
I'll need to go through the source and make that happen.

**Review by dirkhh — CHANGES_REQUESTED** [review]
You are welcome to adjust your PR, but I think it might be easier if I just fix this for all of the builds.

File: core/git-access.c (hunk 1 above)
Comment (flokli) [reply core/git-access.c:362]: There's only 3 occurences - I can update the PR to get rid of the old usages.

**Comment by dirkhh** [comment]
That's part of it.
The other part is fixing the build scripts and cmake dependencies to ensure we find 0.26 or newer...

**Comment by flokli** [comment]
I found some libgit 0.20 comments in android and macos files…

**Comment by flokli** [comment]
Hm, it also seems the `pkg_config_library` abstraction makes it a bit harder to pass a version boundary to `pkg_check_modules` - I think I'll leave that up to you. Thanks!

**Comment by dirkhh** [comment]
OK, I'll take care of it in the next couple of days.
Those comments are part of what needs cleaning up. All the builds that I drive use libgit2 >= 0.26, including Mac and Android. And I believe that at this point all even remotely reasonable Linux distributions ship a version that's newer than that (and the scripts are catching it if they don't - except that those version checks are ALSO broken).
So yeah, this is a slightly bigger task to get right :(

Thanks for bringing it to my attention, that's really useful and appreciated.

**Pull request closed without merging**